source/administration/plugins.rst

```diff
@@ -12,12 +12,63 @@ Consider using a plugin in the following scenarios:
+
+.. image:: https://user-images.githubusercontent.com/915956/66891467-1b18eb80-ef9e-11e9-9de3-37a3c5899bd8.png
+
+Installing a Plugin
+~~~~~~~~~~~~~~~~
+
+When a new plugin becomes available on the marketplace, a new listing with a button labeled **Install** will be displayed.  After clicking **Install**, the latest plugin binary is downloaded from its respective source and installed on the server.    
+
+Configuring and Enabling a Plugin
+~~~~~~~~~~~~~~~
+
+Once a plugin has been installed (or pre-installed if it shipped with the Mattermost binary release) there will be a "Configure" button shown next to the plugin listing.  
+
+1. Click on the button and you will be navigated to the plugins "Settings" page.  
+2. Fill in Plugin settings as required.
+3. At the top of the page, set **Enable Plugin** setting to ``True``. The plugin will remain inactive until this flag is switched on.
+4. Test out the plugin as needed.
+
+Upgrading Plugins
+~~~~~~~~~~~~~~~~~
+
+In v5.16, the Marketplace only supports the installation of new plugins.  To upgrade a plugin, you will need to manually update it by downloading the binary file from the GitHub repository and uploading it in the **System Console > Plugin management** section.  In an upcoming release, the Marketplace will allow you to upgrade a plugin on demand when a new version becomes available.
+
+Marketplace Server
+-----
+
+There are two Marketplace settings in the **System Console > Plugin management** section:
+
+.. image:: https://user-images.githubusercontent.com/915956/66892854-94660d80-efa1-11e9-805c-a85223d43a07.png
+
+- **Enable Marketplace**: Turns the Marketplace User Interface on or off for System Administrators.
+- **Marketplace URL**: The location of the Marketplace Server you want to query for new plugins. Mattermost hosts a Plugin Marketplace for the community and is the default value for this field. You can setup your own Marketplace server if you want to rely on your own infrastructure.
+
+When you are first presented with the Marketplace, your Mattermost server will attempt to contact the Marketplace Server, run by Mattermost, and return a list of available plugins that are appropriate based on the server version that is currently running.  Only your server version is passed over to Mattermost Marketplace. We retain an anonymized record for product analytics whenever a new plugin is installed, unless you have previously opted out of `Telemetry <https://docs.mattermost.com/administration/telemetry.html>`__. The `Plugin marketplace server code <https://github.com/mattermost/mattermost-marketplace>`__ is available as an open-source project and can be used to setup your own private Marketplace if desired.  
```

Comment:
> The `Plugin marketplace server code <https://github.com/mattermost/mattermost-marketplace>`__ is available as an open-source project and can be used to setup your own private Marketplace if desired.  

Should this last line be in the previous paragraph where it talks about setting up your own server? Seems like it would tie better there than next to the telemetry content. 
 
```suggestion
- **Marketplace URL**: The location of the Marketplace Server you want to query for new plugins. Mattermost hosts a Plugin Marketplace for the community and is the default value for this field. You can setup your own Marketplace server if you want to rely on your own infrastructure. The `Plugin marketplace server code <https://github.com/mattermost/mattermost-marketplace>`__ is available as an open-source project and can be used to setup your own private Marketplace if desired.

When you are first presented with the Marketplace, your Mattermost server will attempt to contact the Marketplace Server, run by Mattermost, and return a list of available plugins that are appropriate based on the server version that is currently running.  Only your server version is passed over to Mattermost Marketplace. We retain an anonymized record for product analytics whenever a new plugin is installed, unless you have previously opted out of `Telemetry <https://docs.mattermost.com/administration/telemetry.html>`__.   
```

Reply:
Yes, I think that makes more sense